src/account/operations/transaction/sign_transaction.rs

```diff
@@ -32,6 +35,31 @@ impl AccountHandle {
             stronghold_secret_manager.read_stronghold_snapshot().await?;
         }
 
+        #[cfg(all(feature = "events", feature = "ledger_nano"))]
+        match &*self.secret_manager.read().await {
+            SecretManager::LedgerNano(ledger) | SecretManager::LedgerNanoSimulator(ledger) => {
```

Comment:
Can we do an `if let` i we have an `|`, if yes it reads better than a match with a single branch